dbms/src/Common/tests/gtest_tiflash_security.cpp

```diff
@@ -76,6 +77,53 @@ cert_allowed_cn="tidb"
     ASSERT_EQ((int)new_tiflash_config.allowedCommonNames().count("tidb"), 0);
 }
 
+TEST(TiFlashSecurityTest, EmptyConfig)
+try
+{
+    const auto log = Logger::get();
+
+    for (const auto & c : Strings{
+             // empty strings
+             R"([security]
+ca_path=""
+cert_path=""
+key_path="")",
+             // non-empty strings with space only
+             R"([security]
+ca_path="  "
+cert_path=""
+key_path="")",
+             // only a part of ssl path is set
+             R"([security]
+ca_path="security/ca.pem"
+cert_path=""
+key_path="")",
+             R"([security]
+ca_path=""
+cert_path="security/cert.pem"
+key_path="")",
+             R"([security]
+ca_path=""
+cert_path=""
+key_path="security/key.pem")",
+         })
+    {
+        TiFlashSecurityConfig tiflash_config(log);
+        auto new_config = loadConfigFromString(c);
+        try
+        {
+            tiflash_config.init(*new_config);
+            ASSERT_FALSE(tiflash_config.hasTlsConfig()) << fmt::format("case: {}", c);
+        }
+        catch (Exception & e)
+        {
+            ASSERT_FALSE(tiflash_config.hasTlsConfig()) << fmt::format("case: {}", c);
+            ASSERT_EQ(e.code(), ErrorCodes::INVALID_CONFIG_PARAMETER) << fmt::format("case: {}", c);
+        }
```

Comment:
I think this make the test asserts for each case in the for loop, it is EITHER empty OR invalid. However I suggest a more strict checking for each case, e.g., this is empty, this is invalid, etc.